src/EnergyPlus/FanCoilUnits.cc

```diff
@@ -1938,13 +1941,14 @@ namespace FanCoilUnits {
 			}
 			// obtain unit output with no active heating/cooling
 			Calc4PipeFanCoil( FanCoilNum, ControlledZoneNum, FirstHVACIteration, QUnitOutNoHC, 0.0 );
-			if ( FirstHVACIteration ) {
-				FanCoil( FanCoilNum ).QUnitOutNoHC = QUnitOutNoHC;
-			}
-			else {
+
+			if ( ColdFlowLocked || HotFlowLocked ) {
 				QUnitOutNoHC = FanCoil( FanCoilNum ).QUnitOutNoHC;
+			} else {  // continue to update QUnitOutNoHC while flow is unlocked
+				FanCoil( FanCoilNum ).QUnitOutNoHC = QUnitOutNoHC;
```

Comment:
This seems quite reasonable.  I'll build/test locally, but this makes sense.  My only question is really whether this same catch needs to be extended to other component types.